### PR TITLE
samples: openthread: enable MPSL by default

### PR DESCRIPTION
**Pull request by edmont**
Provide MPSL enabled by default for all OpenThread samples since the alternative configuration does not count with proper radio testing.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

KRKNWK-7696

### CODE REVIEW

**Comment by edmont** [comment]
Pending on https://github.com/nrfconnect/sdk-nrf/pull/3067

**Comment by greg-fer** [comment]
Wouldn't this require an update to the docs as well?
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/openthread/coap_client/README.html

**Comment by edmont** [comment]
> Wouldn't this require an update to the docs as well?
> https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/openthread/coap_client/README.html

Actually nothing to update for the CoAP Client sample. Maybe just adding `CONFIG_MPSL` to this list: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_thread_configuring.html#basic-configuration

**Review by lmaciejonczyk — APPROVED** [review]

**Review by MarekPorwisz — APPROVED** [review]

File: doc/nrf/ug_thread_configuring.rst [hunk 1]
```diff
@@ -22,6 +22,7 @@ To use the Thread protocol in |NCS|, set the following Kconfig options:
 
 * :option:`CONFIG_NETWORKING` - This option enables the generic link layer and the IP networking support.
 * :option:`CONFIG_NET_L2_OPENTHREAD` - This option enables the OpenThread stack required for the correct operation of the Thread protocol and allows you to use them.
+* :option:`CONFIG_MPSL` - Use Nordic Multi Protocol Service Layer (MPSL) implementation, providing services for single and multi-protocol implementations.
```
Comment (greg-fer) [inline doc/nrf/ug_thread_configuring.rst:25]:
```suggestion
* :option:`CONFIG_MPSL` - This options enables the Nordic Multi Protocol Service Layer (MPSL) implementation, which provides services for single-protocol and multi-protocol implementations.
```
Link to OpenThread architectures under `single-protocol and multi-protocol implementations`?
```
:ref:`single-protocol and multi-protocol implementations <ug_thread_architectures>`
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/ug_thread_configuring.rst (hunk 1 above)
Comment (edmont) [reply doc/nrf/ug_thread_configuring.rst]: Thanks, updated.

**Comment by rlubos** [comment]
https://github.com/nrfconnect/sdk-nrf/pull/3067 got merged

**Review by rlubos — APPROVED** [review]

**Comment by edmont** [comment]
We have green CI here now.

**Pull request merged**